### ISSUE TITLE
NR1 search dashboard broken

### ISSUE THREAD

**Issue by austin-schaefer** [newrelic/docs-website#2598]
### Description

The staging dashboard **Docs Site Instrumentation Dashboard > Search Usage** doesn't work properly since the big search update. Number of searches and top searches are broken.

### Steps to reproduce

1. Go to the dashboard
2. Look at the results

### Expected

- Dashboards return results

**Comment by jpvajda** [comment]
query> Top Searches over past week

`SELECT count(*) From PageAction where actionName = 'swiftypeSearch_input' facet searchTerm since 1 week ago`

query> Number of searches over the past week

`SELECT count(*) as 'Number of Searches' From PageAction where actionName = 'swiftypeSearch_input' timeseries since 1 week ago`

**Comment by roadlittledawn** [comment]
i wonder if we forgot to add `PageAction` tracking in the [recent search UI update](https://github.com/newrelic/gatsby-theme-newrelic/pull/375).

**Comment by nr-kkenney** [comment]
got the first PR out for this. just a heads up, based on some very simple local testing, the new search experience + the way we implement tracking the search may result in "awkward" data being captured. this stems from the search experience being immediately reactive to what the user types. 

For example:
if i want to search for the word "infrastructure" the following was captured as data:
```none
infra
infrastruc
infrastructure
```
this is due to the debounce we specified and my typing speed. but 

the tl;dr - if a user is slow to type, we will likely get data for part of their search, in addition to the complete search (once they stop typing).

**Comment by jpvajda** [comment]
@nr-kkenney interesting.. the reactive search is definitely an awesome experience, and by capturing each input we do get a sense of where a user would "stop" typing based on the results they are seeing, we could then use this data to refine swiftype and perhaps even consider weighting results higher based on this data we are capturing. So I don't think it's an issue, as we are instrumenting a reactive search expeirence. it may come down to how the data is actually analyzed and used IMHO

**Comment by nr-kkenney** [comment]
@jpvajda i didnt give a bad enough example 🙈 

i think the case that is not so valuable is something like if i search for the word "cat", worst case scenario is we see data for:
```none
c
ca
cat
```
so rather than just seeing the user search for cat, if the user types slow enough, we see things they didnt search for.

**Comment by jpvajda** [comment]
@nr-kkenney 😄 I guess as long as we capture the full search string, along with the substrings, the data could be filtered...is there a way to better control the firing of events so we only pass an event after a full string is entered? I guess it's hard to know as a user may stop at any point in a reactive search queries as they may find what they are looking for...

`i..n..f..r...a `>>> finds results,
or 
`i..n..f..r..a..s..t..r `>>> finds results.

**Comment by nr-kkenney** [comment]
clinton & I talked about increasing the debounce time, giving the user more time to type, but that also decreases how reactive the search experience is. 

i didnt want to problem solve this or even push too hard on this (although it may sound like i am doing this above 🙈 ), since it may be fine overall once we see the data.

an idea that comes to mind though is that we only fire the tracking event for the user's search if they click into a result, or some other final action. something that signals to us the user is done searching.

**Comment by austin-schaefer** [comment]
Firing just after click would be an issue I think if they don’t click. Those searches with no clicks are often the most interesting, knowing what users are NOT finding good answers on. 

But perhaps some combination? EG either when they click, or after a short lag with no characters entered (less than a second)?

**Comment by jpvajda** [comment]
I feel like we are over rotating on this particular concern given the reactive nature of the search. the main goal here is to fix the dashboard reporting so that search data is once again available to a user for analysis.  From there, if the data is just completely non useful, we can discuss if there is something better to do , perhaps with a more refined NRQL query. 

Have we not yet resolved this original problem of no data being reported @nr-kkenney ?

**Comment by nr-kkenney** [comment]
all sites should be updated & changes pulled in. looks like the dashboard has data in it again, closing this ticket.
